_posts/2021-01-26-converting-puppetized-dsc-modules.md

```diff
@@ -0,0 +1,154 @@
+If there’s a problem calling the DSC resources from Puppet, we’ll work on it, but if there’s a problem with how those resources behave, that’s upstream.
+
+**Seriously though what is there to not love?**
+
+* Minimal module maintainence costs
+* Automated module building
+* Property by property reporting
+* VSCode Extension with intellisense
+* We have 68 modules on the Forge in the [DSC namespace][dsc-namespace]
+* Converting is simple
+
+## So how can I do this?
+*Old method:* 
+```
+mod 'puppetlabs-dsc', '1.9.4'
+```
+
+*New method:*
+```
+mod 'dsc-activedirectorycsdsc', '3.1.0.0'
+mod 'dsc-auditpolicydsc', '1.4.0.0'
+mod 'dsc-certificatedsc', '4.4.0.0'
+mod 'dsc-computermanagementdsc', '6.2.0.0'
+mod 'dsc-dfsdsc', '4.3.0.0'
+mod 'dsc-networkingdsc', '7.0.0.0'
+mod 'dsc-officeonlineserverdsc', '1.2.0.0'
+mod 'dsc-securitypolicydsc', '2.7.0.0'
+mod 'dsc-sharepointdsc', '3.2.0.0'
+mod 'dsc-sqlserverdsc', '12.3.0.0'
+mod 'dsc-storagedsc', '4.5.0.0'
+mod 'dsc-systemlocaledsc', '1.2.0.0'
+mod 'dsc-xactivedirectory', '2.24.0.0'
+mod 'dsc-xazure', '0.2.0.0'
+mod 'dsc-xazurepack', '1.4.0.0'
+mod 'dsc-xbitlocker', '1.4.0.0'
+mod 'dsc-xcredssp', '1.3.0.0'
+mod 'dsc-xdatabase', '1.9.0.0'
+mod 'dsc-xdefender', '0.2.0.0'
+mod 'dsc-xdhcpserver', '2.0.0.0'
+mod 'dsc-xdisk', '1.0'
+mod 'dsc-xdismfeature', '1.3.0.0'
+mod 'dsc-xdnsserver', '1.11.0.0'
+mod 'dsc-xexchange', '1.27.0.0'
+mod 'dsc-xfailovercluster', '1.12.0.0'
+mod 'dsc-xhyper-v', '3.16.0.0'
+mod 'dsc-xinternetexplorerhomepage', '1.0.0'
+mod 'dsc-xjea', '0.3.0.0'
+mod 'dsc-xmysql', '2.1.0.0'
+mod 'dsc-xpendingreboot', '0.4.0.0'
+mod 'dsc-xphp', '1.2.0.0'
+mod 'dsc-xpowershellexecutionpolicy', '3.1.0.0'
+mod 'dsc-xpsdesiredstateconfiguration', '8.5.0.0'
+mod 'dsc-xremotedesktopadmin', '1.1.0.0'
+mod 'dsc-xremotedesktopsessionhost', '1.8.0.0'
+mod 'dsc-xrobocopy', '2.0.0.0'
+mod 'dsc-xscdpm', '1.2.0.0'
+mod 'dsc-xscom', '1.3.3.0'
+mod 'dsc-xscsma', '2.0.0.0'
+mod 'dsc-xscspf', '1.3.1.0'
+mod 'dsc-xscsr', '1.3.0.0'
+mod 'dsc-xscvmm', '1.2.4.0'
+mod 'dsc-xsmbshare', '2.1.0.0'
+mod 'dsc-xsqlps', '1.4.0.0'
+mod 'dsc-xtimezone', '1.8.0.0'
+mod 'dsc-xwebadministration', '2.5.0.0'
+mod 'dsc-xwebdeploy', '1.2.0.0'
+mod 'dsc-xwindowseventforwarding', '1.0.0.0'
+mod 'dsc-xwindowsrestore', '1.0.0'
+mod 'dsc-xwindowsupdate', '2.7.0.0'
+mod 'dsc-xwineventlog', '1.2.0.0'
+mod 'dsc-xwordpress', '1.1.0.0'
+```
+
+I know this looks like a lot of work, but really the work has been done for you! Just copy and paste the above code snippet. 
+
+> The modules above are alike to the *puppetlabs-dsc* module as we could possibly get. 
+> It is strongly recommended to migrate to these versions and when everything is up and running, upgrading to the latest versions.
+
+## A few caveats
+
+The following modules cannot be automatically puppetized at the same version because they referenced commit SHAs instead of versions discoverable on the [gallery;][psgallery] instead release the next stable version after these SHA's. 
```

Comment:
```suggestion
The following modules cannot be automatically puppetized at the same version because they referenced commit SHAs instead of versions discoverable on the [gallery;][psgallery] instead, we recommend using the next released stable version after these SHA's, as listed below:
```